### PR TITLE
add message link, resolve recursion, switch to display names

### LINKED ISSUE #3: Use guild nicknames instead of usernames
Use guild nickname instead of Discord username in TL;DR response prompt for better user experience and contextual understanding.

TL;DR: Let's opt for Discord guild nickname over Discord username.

[Interaction Data](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-interaction-data)
[Guild Member Object](https://discord.com/developers/docs/resources/guild#guild-member-object)
[Message Object](https://discord.com/developers/docs/resources/channel#message-object)

### LINKED ISSUE #5: TL;DR should include original message link
At present, the TL;DR feature lacks a reference to the specific message being summarized. To improve the clarity and usefulness of TL;DR summaries, it would be beneficial to include a link to the original message within the TL;DR output. This addition would provide users with easy access to the complete context and facilitate efficient navigation within the conversation.

### LINKED ISSUE #7: Handle recursive TL;DRs
> It hurt itself in its confusion!

Currently, the TL;DR function has the potential to generate a summary of its own message, which results in redundancy and confusion in the LLM (Language Model). To address this problem, we propose implementing a verification process that compares the ID of the author of the target message with the bot's ID. By excluding the bot's ID from the TL;DR process, we can avoid self-TL;DR and enhance the overall effectiveness of this feature.

### Mitigation

To prevent self-TL;DR and improve the effectiveness of the TL;DR function, we suggest implementing a verification process. This process would involve comparing the ID of the author of the target message with the bot's ID. By excluding the bot's ID from the TL;DR process, we can ensure that the function doesn't generate a summary of its own message. This will eliminate redundancy and confusion in the LLM.

TL;DR: The bot should not attempt to TL;DR itself.

### PR DESCRIPTION
**Pull request by jjoeldaniel**
* Resolves #3
* Resolves #5 
* Resolves #7

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -94,7 +109,7 @@ export async function handle(request: Request): Promise<Response> {
             botID: env.DISCORD_CLIENT_ID,
             botToken: env.DISCORD_TOKEN,
             interactionToken: interaction.token,
-            content: `TL;DR: ${result}`,
+            content: `TL;DR: ${result} \n\n${messageURL}`,
```
Comment (EthanThatOneKid) [inline main.ts]:
```suggestion
            content: `TL;DR: ${result} \n\n↩${messageURL}`,
```

File: bot/discord/discord_api_client.ts
Comment (EthanThatOneKid) [inline bot/discord/discord_api_client.ts:1]: Nit/suggestion:
```ts
import type { discord } from "../../deps.ts";
import type {
  DiscordAPIClientInterface,
  EditOriginalInteractionResponseOptions,
  RegisterCommandOptions,
  RetrieveGuildUserOptions,
} from "./discord_api_client_interface.ts";
```

File: main.ts (hunk 1 above)
Comment (EthanThatOneKid) [reply main.ts]: Do you like how this looks in qa?

Comment (jjoeldaniel) [reply main.ts]: image

Not bad

**Review by EthanThatOneKid — APPROVED** [review]
Nice work! Next time, please consider making a separate PR per issue[^1]. 

[^1]: https://google.github.io/eng-practices/review/developer/small-cls.html

**Pull request merged**